### ISSUE TITLE
unpaired surrogate in OsStr on Windows leads to debug_assert failure in v6.4.0

### ISSUE THREAD

**Issue by oconnor663** [dylni/os_str_bytes#14]
My `b3sum` crate depends on `os_str_bytes` (6.4.0) indirectly via `clap` (4.0.26), and I have a test case that uses invalid UTF-16 filenames on Windows. https://github.com/dylni/os_str_bytes/commit/76c28d5e469c2627e9881fa5bccd6858534ac383#diff-3b7ba22f3645054b5b1c07651b4c378e01c0f8eda295e24e4bedf4e6e3a79382R72 added a `debug_assert` that is [failing in CI for me](https://github.com/BLAKE3-team/BLAKE3/actions/runs/3509711746/jobs/5878999304#step:43:108). The following is a minimized repro of what I think is causing the crash:

```rust
use os_str_bytes::RawOsStr;
use std::ffi::OsString;
use std::os::windows::ffi::OsStringExt;

fn main() {
    // Create an invalid UTF-16 string containing a single "unpaired surrogate".
    let bad_str = OsString::from_wide(&[0xDC00]);
    let raw = RawOsStr::new(&bad_str);
    dbg!(raw.as_raw_bytes());
    // This crashes in debug mode with os_str_bytes 6.4.0 but not 6.3.1.
    raw.to_os_str();
}
```

**Comment by dylni** [comment]
Thanks for this issue! This has been fixed in [version 6.4.1](https://github.com/dylni/os_str_bytes/releases/tag/6.4.1), but I'm surprised this wasn't caught in fuzz testing I performed before the release.

The previous version has been yanked as well. However, this should not have caused any incorrect results, aside from the failed assertion, for any Rust version so far.

**Comment by oconnor663** [comment]
Confirmed that v6.4.1 fixes my tests. Thanks for the lightning speed fix!
